### PR TITLE
4910 Add the mean iou handler

### LINKED ISSUE #4910: Make the class name of the IoU metric consistent with other metrics
**Is your feature request related to a problem? Please describe.**
The class name of IoU metrics is "MeanIoU", which is not consistent with other metrics, like the [dice metric](https://github.com/Project-MONAI/MONAI/blob/535c6f1fc8767c09a21dba2464ca687f23c85154/monai/metrics/meandice.py#L22).  And it is similar to a metric handler, like the [mean dice handler](https://github.com/Project-MONAI/MONAI/blob/535c6f1fc8767c09a21dba2464ca687f23c85154/monai/handlers/mean_dice.py#L19). This may make users confuse about the IoU handler and metric.

**Describe the solution you'd like**

- [ ] Change the class name of the iou metric to "IoUMetric"
- [ ] Add a "mean_iou.py" file to the handler folder, in which an IoU handler named "MeanIoU" will be implemented.

@Nic-Ma @wyli Do you think if it's ok to change like this?

### PR DESCRIPTION
**Pull request by binliunls**
Signed-off-by: binliu <binliu@nvidia.com>

Fixes #4910 .

### Description
The IoU metric "MeanIoU" lacks a corresponding handler, which will lead to inconvinience when using this metric in ignite engine as key_metric. Therefore, I'd like to add a handler of IoU metric named "MeanIoUHandler" to fix this problem. 

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: tests/test_handler_mean_iou.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    (4, 2),
+]
+
+
+class TestHandlerMeanIoU(unittest.TestCase):
+    # TODO test multi node averaged iou
+
+    @parameterized.expand([TEST_CASE_1, TEST_CASE_2, TEST_CASE_3])
+    def test_compute(self, input_params, expected_avg, details_shape):
+        dice_metric = MeanIoUHandler(**input_params)
```
Comment (Nic-Ma) [inline tests/test_handler_mean_iou.py:34]: Please change the variable name, should not be "dice metric".
Same for several other cases in this file.

Thanks.

Comment (binliunls) [reply tests/test_handler_mean_iou.py]: Thanks Nic,
The name has been changed.

**Review by Nic-Ma — APPROVED** [review]
Thanks for the quick update.
Looks good to me now.

**Pull request merged**